pyproject.toml

```diff
@@ -1,54 +1,93 @@
 [build-system]
-requires = [
-    "scikit-build>=0.12",
-    "setuptools>=42",
-    "setuptools-scm[toml]",
+requires = ["scikit-build-core"]
+build-backend = "scikit_build_core.build"
+
+[project]
+name = "cmake"
+description = "CMake is an open-source, cross-platform family of tools designed to build, test and package software"
+keywords = ["CMake", "build", "c++", "fortran", "cross-platform", "cross-compilation"]
+readme = "README.rst"
+license = {text = "Apache 2.0"}
+authors = [
+    {name = "Jean-Christophe Fillion-Robin", email = "jchris.fillionr@kitware.com"},
+]
+classifiers = [
+    "License :: OSI Approved :: Apache Software License",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: C",
+    "Programming Language :: C++",
+    "Programming Language :: Fortran",
+    "Programming Language :: Python",
+    "Operating System :: OS Independent",
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development :: Build Tools",
+    "Typing :: Typed"
+]
+dynamic = ["version"]
+requires-python = ">=3.7"
+
+[project.urls]
+Homepage = "https://cmake.org"
+Documentation = "https://cmake-python-distributions.readthedocs.io"
+Source = "ttps://github.com/scikit-build/cmake-python-distributions"
+"Mailing list" = "https://groups.google.com/forum/#!forum/scikit-build"
+"Bug Tracker" = "https://github.com/scikit-build/cmake-python-distributions/issues"
+
+[project.optional-dependencies]
+test = [
+  "coverage>=4.2",
+  "importlib_metadata>=4.6; python_version<'3.10'",
+  "pytest>=3.0.3",
+  "pytest-cov>=2.4.0",
 ]
-build-backend = "setuptools.build_meta"
+
+[project.scripts]
+cmake = "cmake:cmake"
+cpack = "cmake:cpack"
+ctest = "cmake:ctest"
+
+[tool.scikit-build]
+minimum-version = "0.4"
+build-dir = "build/{wheel_tag}"
+metadata.version.provider = "scikit_build_core.metadata.setuptools_scm"
+ninja.make-fallback = false
+sdist.include = ["src/cmake/_version.py"]
+wheel.py-api = "py2.py3"
```

Comment:
This is not in line with `requires-python = ">=3.7"` and I think pip will not install the wheel from PyPI when running python 2.7/3.6.
Can/Shall we drop/update the `requires-python` ?
If we still support 2.7/3.6 then they shall still be tested.

Reply:
Python 2.7 & 3.5 can't use the wheel produced. I'm checking what can be done.

Reply:
I opened #455 depending on python<3.7 support requirement (for wheels).